### PR TITLE
Bkit 161 mask subject other bookings

### PR DESCRIPTION
**Pull request by aochsner**
Also bumped kotlin version

### CODE REVIEW

File: src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt [hunk 1]
```diff
@@ -67,14 +72,21 @@ class BookableController(private val bookableRepository: BookableRepository, pri
...
             .filter { it.locationId == locationId }
             .map { bookable ->
                 val bookings = when {
                     "bookings" in expand ?: emptyList() ->
-                        bookingRepository.getAllBookings()
+                        allBookings
                             .filter { booking -> booking.bookableId == bookable.id }
                             .filter { desiredInterval.overlaps(it.interval(timeZone)) }
+                            .map { booking ->
+                                if (booking.user.externalId == user.subject)
```
Comment (aochsner) [inline src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt]: is comparing the externalId good enough here so i don't have to go back to the database?  If so, probably could do the same for DELETE /booking

File: src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt [hunk 2]
```diff
@@ -67,14 +72,21 @@ class BookableController(private val bookableRepository: BookableRepository, pri
...
+                                    booking
+                                else
+                                    booking.copy(subject = MASKED_STRING)
```
Comment (aochsner) [inline src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt]: masking OK?

File: build.gradle [hunk 3]
```diff
@@ -40,8 +40,8 @@ buildscript {
 
 plugins {
     id 'org.springframework.boot' version '1.5.9.RELEASE'
-    id 'org.jetbrains.kotlin.jvm' version '1.2.0'
-    id 'org.jetbrains.kotlin.plugin.spring' version '1.2.0'
+    id 'org.jetbrains.kotlin.jvm' version '1.2.10'
+    id 'org.jetbrains.kotlin.plugin.spring' version '1.2.10'
```
Comment (aochsner) [inline build.gradle]: minor, unrelated but wanted to

File: src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt (hunk 1 above)
Comment (mathomas) [reply src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt]: Seems good enough to me, and natural in the context.  But why did we bother creating a local user ID and mapping it, in that case?  Not a challenge to you, but more food for thought.

File: src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt (hunk 2 above)
Comment (mathomas) [reply src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt]: Aye-aye.

File: src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt (hunk 1 above)
Comment (aochsner) [reply src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt]: well, we need to be able to show who has a bookable booked.  So you either make that part of the booking table or normalize it a bit to a users table... i still think that's the right call... and if you do that, then it makes more sense to have an internal id be the PK/FK...but the "business key" is still the external id... plus it probably helps when we add admin users and/or "limit to only *these* users"

Comment (mathomas) [reply src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt]: OK, local ID more for capturing relationships than necessarily for selection, might be a way to say it?  I'm good with it.  I think ;-)

**Pull request merged**